### PR TITLE
Fixed deadlock bug, cleaned code and some minor fixes.

### PR DESCRIPTION
**Pull request by emilwestergren**

### CODE REVIEW

File: source/draw/gpu/opengl/OpenGLES3/Vao.ooc [hunk 1]
```diff
@@ -29,7 +29,7 @@ Vao: class {
 	bind: func { glBindVertexArray(backend) }
 	unbind: func { glBindVertexArray(0) }
 	_generate: func (positions: Float*, textureCoordinates: Float*, vertexCount: UInt, dimensions: UInt) -> Bool {
-		version(debugGL){ Debug print("Allocating OpenGL VAO") }
+		version(debugGLv){ Debug print("Allocating OpenGL VAO") }
```
Comment (davidhesselbom) [inline source/draw/gpu/opengl/OpenGLES3/Vao.ooc]: Space plz `) {`

File: source/draw/gpu/opengl/OpenGLES3/Context.ooc [hunk 2]
```diff
@@ -139,12 +139,12 @@ Context: class {
...
 	}
 	create: static func ~shared (window: NativeWindow, sharedContext: This = null) -> This {
-		Debug print("Creating OpenGL context")
+		version(debugGLv){ Debug print("Creating OpenGL context") }
 		result := This new()
 		result _generate(window, sharedContext) ? result : null
 	}
 	create: static func ~pbufferShared (sharedContext: This = null) -> This {
-		Debug print("Creating OpenGL context")
+		version(debugGLv){ Debug print("Creating OpenGL context") }
```
Comment (davidhesselbom) [inline source/draw/gpu/opengl/OpenGLES3/Context.ooc]: Space plz `) {`

File: source/draw/gpu/opengl/OpenGLES3/Context.ooc [hunk 3]
```diff
@@ -139,12 +139,12 @@ Context: class {
 		return result
 	}
 	create: static func ~shared (window: NativeWindow, sharedContext: This = null) -> This {
-		Debug print("Creating OpenGL context")
+		version(debugGLv){ Debug print("Creating OpenGL context") }
```
Comment (davidhesselbom) [inline source/draw/gpu/opengl/OpenGLES3/Context.ooc]: Space plz `) {`

**Comment by emilwestergren** [comment]
Rebased and updated.

File: source/draw/gpu/opengl/OpenGLES3/Context.ooc [hunk 4]
```diff
@@ -139,12 +139,12 @@ Context: class {
...
 	}
 	create: static func ~shared (window: NativeWindow, sharedContext: This = null) -> This {
-		Debug print("Creating OpenGL context")
+		version (debugGLv) { Debug print("Creating OpenGL context") }
 		result := This new()
 		result _generate(window, sharedContext) ? result : null
 	}
 	create: static func ~pbufferShared (sharedContext: This = null) -> This {
-		Debug print("Creating OpenGL context")
+		version (debugGLv) { Debug print("Creating OpenGL context") }
```
Comment (davidhesselbom) [inline source/draw/gpu/opengl/OpenGLES3/Context.ooc]: `debugGLv` should be more verbose.

Comment (davidhesselbom) [inline source/draw/gpu/opengl/OpenGLES3/Context.ooc]: `version (flag)` with space breaks syntax highlighting, so that's probably not right.

**Comment by emilwestergren** [comment]
Updated.

File: source/draw/gpu/opengl/OpenGLES3/Vao.ooc [hunk 5]
```diff
@@ -29,7 +29,7 @@ Vao: class {
 	bind: func { glBindVertexArray(backend) }
 	unbind: func { glBindVertexArray(0) }
 	_generate: func (positions: Float*, textureCoordinates: Float*, vertexCount: UInt, dimensions: UInt) -> Bool {
-		version(debugGL){ Debug print("Allocating OpenGL VAO") }
+		version (debugGL) { Debug print("Allocating OpenGL VAO") }
```
Comment (davidhesselbom) [inline source/draw/gpu/opengl/OpenGLES3/Vao.ooc]: here

File: source/draw/gpu/android/GpuPackerBin.ooc [hunk 6]
```diff
@@ -47,5 +54,10 @@ GpuPackerBin: class {
 			arrayList removeAt(index, false)
 		result
 	}
-	find: func (size: IntSize2D, bytesPerPixel: UInt)-> GpuPacker { this _search(size, bytesPerPixel, this _packers) }
+	find: func (size: IntSize2D, bytesPerPixel: UInt)-> GpuPacker {
```
Comment (davidhesselbom) [inline source/draw/gpu/android/GpuPackerBin.ooc]: space before `->`

**Pull request merged**